modules/ROOT/pages/clauses/where.adoc

```diff
@@ -486,18 +486,25 @@ The `name` and `age` for `Andy` are returned because his name starts with 'AND'
 [[filter-on-patterns]]
 === Filter on patterns
 
-Patterns are expressions in Cypher, expressions that return a list of paths.
-List expressions are also predicates; an empty list represents `false`, and a non-empty list represents `true`.
-
-Patterns are not only expressions, they are also predicates.
-The only limitation to a pattern is that it must be able to express it in a single path.
-Unlike `MATCH`, it is not possible to use commas between multiple paths.
-To combine multiple patterns using `WHERE`, instead use `AND`.
-Note that new variables cannot be introduced.
-Although it might look very similar to the `MATCH` patterns, the `WHERE` clause is used to eliminate matched paths.
-`MATCH (a)-[*]->(b)` is very different from `WHERE (a)-[*]->(b)`.
-The first will produce a path for every path it can find between `a` and `b`, whereas the latter will eliminate any matched paths where `a` and `b` do not have a directed relationship chain between them.
+Path patterns may be used both in a MATCH clause and in a WHERE clause. When used in a WHERE clause they are called path
+pattern expressions.
 
+Path patterns used in a MATCH clause should generally be thought of as retrieving data. Path patterns expressions used
+in a WHERE clause should on the other hand be thought of as eliminating data. The clause `MATCH (a)-[]->(b)` is very
```

Comment:
I find "eliminating" sounds a lot like we are `DELETE`ing. The purpose of `WHERE` in general is to _filter_  the results/data that we found with `MATCH`.

What do you think about contrasting retrieving (or maybe "enumerating") in `MATCH` with "testing"  in `WHERE`? One could probably also use/promote the `EXISTS { ... }` or `exists(...)` expression here to explain what the pattern does in `WHERE`.

Reply:
I agree that "eliminating" was a poor choice of words. I think that "retrieving" and "filtering" are good though. The page already uses "filtering" in many places which means the user should have a grasp of the concept by the time he reaches this section.

Reply:
Both Jens and you have offered some suggestions about linking to other parts of the documentation which have their own sections in the docs. It would feel a bit spaghetti to me to include something about those other sections in here, so I may extract the "path pattern expressions" to their section of the docs and then link to them from relevant places such as WHERE (this page), CASE WHEN (other page), exists function (other page). I would not mention the EXISTS subquery as that has no special handling of path pattern expressions - if the EXISTS subquery has a WHERE clause then it can accept a path pattern expression, but so can anything else with a WHERE clause.

Reply:
I feel that using `EXISTS { p }` or `exists( p )` in here will be quite confusing for users. This is because `EXISTS { }` accepts a path pattern whereas `exists( )` where the parameter of the function is not a boolean but we accept a path pattern expression anyways (this is the only exception to the rule).

Reply:
I like "filtering". That goes well together with the rest of the text.
What I meant with the `EXISTS { ... }` expression is that `WHERE (a)-[]→(b)` can be thought of as a shorthand of `WHERE EXISTS { (a)-[]→(b) }`.

I guess, pattern expressions are independent enough from `WHERE` clauses that they may reside in a page of their own. But @JPryce-Aklundh probably has an opinion on extracting that section.

Reply:
You are right, `exists ( pattern-expression )` is nothing we want to promote. We mostly keep it as a legacy and to not unnecessarily introduce breaking changes.

I am not sure I understand your point about `EXISTS { ... }`, though: `EXISTS { ... }` is the more powerful construct and can accept arbitrary queries, including what can be expressed as path pattern expression. My idea was to frame path pattern expression as a short-hand to this. That way we promote `EXISTS { ... }` as a means when people want to express more than what can be expressed in a path pattern expression. And if people know `EXISTS { ... }` already, then such an equivalence will make it easier for them to learn path pattern expressions.

Reply:
This is just an idea, of course, so feel free to ignore it.

Reply:
> I guess, pattern expressions are independent enough from WHERE clauses that they may reside in a page of their own. But @JPryce-Aklundh probably has an opinion on extracting that section.

I am going to try to make it work by keeping it on the current page. Other options I considered was moving it to the ["Cypher Expressions"](https://neo4j.com/docs/cypher-manual/current/queries/expressions/) page, or somewhere in the ["Cypher Patterns"](https://neo4j.com/docs/cypher-manual/current/patterns/) directory.

All the options have pros and cons. I found it hard to get a clean result.

This is perhaps more of a comment for Jens, but I found that generally the docs were not structure the way I personally think of Cypher. In the long term I would imagine that we could have something like this (the devil would be in the details though...)

- **Expressions** page which talks specifically about the different expressions you can have (variable reference, property access, has type, has label, booleans, math operators, functions, procedures, ...). This is not a page which currently exists and it's a bit all over the place ("CASE expression" has its own page, some expressions live in the "Cypher Expressions" page, some are jumbled in the "WHERE" page, and some are in the "Values and Types" pages).
- **Where** page which talks about specifically where this clause can be used and which keywords can be combined with it (IN, NOT, STARTS WITH, ENDS WITH, ...)